### PR TITLE
Add spell check GHA and clean up spelling

### LINKED ISSUE #110: Add spell check GHA
We should add this GHA to run for all PRs to `main`, using the R script approach (which I personally like more since I can run the script locally, but if someone strongly disagrees let me know here!).

This will involve..

- Add `components` directory as in https://github.com/AlexsLemonade/training-modules/tree/72621ada8ecffc524e7fbbee78172dca9d0f5c85/components
- Add `scripts/spell-check.R` as in https://github.com/AlexsLemonade/training-modules/blob/72621ada8ecffc524e7fbbee78172dca9d0f5c85/scripts/spell-check.R
- Add `.github/workflows/spell-check.yml` as in https://github.com/AlexsLemonade/training-modules/blob/master/.github/workflows/spell-check.yml

### PR DESCRIPTION
**Pull request by sjspielman**
**⚠️ Stacked on #122**
Closes #110 

This PR adds a spell check GHA to run on PRs to `main`, and **for now**, also to `template-refresh` (we'll remove this as part of #130).
Along the way, I fixed a variety of spelling errors and populated the dictionary. On some occasions, I found some spelling errors that are going to go away with forthcoming development, such as `htmlpreview` or `CCDL`. Rather than adding these to the dictionary, I went ahead and removed those words, eg `CCDL` -> `Data Lab`.

Note to self:
**_Do not merge this into a stacked branch!! Keep this open and wait to merge into `template-refresh` so the GHA gets tested_.** (we could test the GHA in other ways, but imo this seems cleanest).

### CODE REVIEW

File: components/dictionary.txt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+customizable
+Danio
+DCP
+DESeq
+DESeqDataset
+discoverable
+dropdown
+DropSeq
+ecure
+ecursively
```
Comment (jashapiro) [inline components/dictionary.txt:33]: Love it (I know why these are here)

File: setup-issue-templates/draft-schedule.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Update `workshop/SCHEDULE.md` to point to appropriate materials for your workshop.
 
   - Add relative links to PDFs in `slides` to the schedule.
-  - We recommend using [htmlpreview](https://github.com/htmlpreview/htmlpreview.github.com) to display rendered versions of R Notebooks that are tied to a specific release in the [`AlexsLemonade/training-modules`](https://github.com/AlexsLemonade/training-modules). You can use the `{{site.release_tag}}` convention used throughout the template.
+  - Use relative links to display rendered versions of R Notebooks in `completed-notebooks`.
```
Comment (jashapiro) [inline setup-issue-templates/draft-schedule.md:4]: Probably for the other PR, but instructions for getting the rendered notebooks into this repo?

File: software-setup/slack-procedures.md [hunk 3]
```diff
@@ -108,7 +108,7 @@ First, use the new message button in the top right side corner of the Slack inte
 
 <img src = "screenshots/slack-compose-new-message.png" width="300">
 
-You are then able to search for the Data Lab instructors – `Chante Bethell`, `Ally Hawkins`, `Jen O'Malley`, `Josh Shapiro`, or `Stephanie Spielman` – and compose your message.
+You are then able to search for the Data Lab instructors and compose your message.
```
Comment (jashapiro) [inline software-setup/slack-procedures.md:111]: Hmmm... maybe we want to make this a list in config and fill these names in automatically? Do we name ourselves elsewhere that we would reuse this?

File: software-setup/slack-procedures.md [hunk 4]
```diff
@@ -133,6 +133,6 @@ Or you can type three backticks <code>```</code> into the message field and form
 
 <img src = "screenshots/slack-backticks.gif" width = "600">
 
-You are then ready to paste your code or error message into the grey code block.
+You are then ready to paste your code or error message into the gray code block.
```
Comment (jashapiro) [inline software-setup/slack-procedures.md:136]: Boo. American English has this wrong.

File: components/dictionary.txt [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+Muris
+musculus
+nonconsensual
+OSCA
+PHI
+PII
+pre
+README
+rerio
+rstudio
```
Comment (jashapiro) [inline components/dictionary.txt:70]: Do we want the uncapitalized version?

File: .github/workflows/spell-check.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Run spell check
+        id: spell_check_run
+        run: |
+          results=$(Rscript --vanilla "scripts/spell-check.R")
+          echo "sp_chk_results=$results" >> $GITHUB_OUTPUT
+          cat spell_check_errors.tsv
+
+      - name: Archive spelling errors
+        uses: actions/upload-artifact@v3
+        with:
```
Comment (jashapiro) [inline .github/workflows/spell-check.yml:35]: Probably not needed, but why not.
```suggestion
        if: ${{ steps.spell_check_run.outputs.sp_chk_results > 0 }}
        with:
```

**Review by jashapiro — APPROVED** [review]
LGTM

My suggestions here are all pretty minor, so I don't need to see this again. (The one larger one is adding instructor names as a config variable, but that should probably happen separately if we want to do it.)

File: software-setup/slack-procedures.md (hunk 4 above)
Comment (sjspielman) [reply software-setup/slack-procedures.md:136]: for the record I agree!

File: setup-issue-templates/draft-schedule.md (hunk 2 above)
Comment (sjspielman) [reply setup-issue-templates/draft-schedule.md:4]: Yeah, I plan to circle back to all of this as part of #112, but can definitely get it started in the other GHA PR if you prefer

File: components/dictionary.txt (hunk 5 above)
Comment (sjspielman) [reply components/dictionary.txt]: Went back and looked at how this got in there. Fixed! https://github.com/AlexsLemonade/training-specific-template/pull/131/commits/40088beaee45fd8929ffc7006cc51ecf36cc7e5f

File: software-setup/slack-procedures.md (hunk 3 above)
Comment (sjspielman) [reply software-setup/slack-procedures.md:111]: This is the only place according to `grep`, but I'll make a note of this over in #106 in case

**Comment by sjspielman** [comment]
**No merging until the base is `main`**!

**Comment by sjspielman** [comment]
spell check GHA works :tada:

**Pull request merged**